mkosi/__init__.py

```diff
@@ -3466,12 +3456,21 @@ def invoke_repart(
 
     output = json.loads(run(cmdline, stdout=subprocess.PIPE).stdout)
 
+    roothash = usrhash = None
     for p in output:
-        if p["type"].startswith("usr") or p["type"].startswith("root"):
-            usr_only = p["type"].startswith("usr")
-            return (p["label"], p.get("roothash"), usr_only)
+        if (h := p.get("roothash")) is None:
+            continue
+
+        if not p["type"].startswith("usr") or p["type"].startswith("root"):
+            die(f"Found roothash property on unexpected partition type {p['type']}")
+
+        # When there's multiple verity enabled root or usr partitions, the first one wins.
+        if p["type"].startswith("usr"):
+            usrhash = usrhash or h
+        else:
+            roothash = roothash or h
 
-    return (None, None, False)
+    return f"roothash={roothash}" if roothash else f"usrhash={usrhash}"
```

Comment:
If we find neither, this should probably not return `usrhash=None` or am I missing something?
```python
            # having both root and usr verity is not supported, always prefer root
            if p["type"].startswith("root"):
                return f"roothash={h}"
            # always prefer the first verity partition we find
            usrhash = usrhash or h
            
    return f"usrhash={usrhash}" if usrhash or None

Reply:
Oops